src/modules/profile/pages/profileSettings/components/CreatorNotifications.tsx

```diff
@@ -0,0 +1,180 @@
+import { useMutation, useQuery } from '@apollo/client'
+import { Avatar, FormControl, FormLabel, HStack, Select, Switch, VStack } from '@chakra-ui/react'
+import { useState } from 'react'
+import { useParams } from 'react-router'
+
+import { MUTATION_UPDATE_CREATOR_NOTIFICATIONS_SETTINGS } from '@/modules/profile/graphql/mutations/creatorNotificationMutation'
+import { QUERY_CREATOR_NOTIFICATIONS_SETTINGS } from '@/modules/profile/graphql/queries/creatorNotificationSettingsQuery'
+import { Body } from '@/shared/components/typography'
+
+type NotificationConfiguration = {
+  id: string
+  name: string
+  value: string
+  type: string
+  options?: string[]
+}
+
+type NotificationSettings = {
+  notificationType: string
+  isEnabled: boolean
+  configurations: NotificationConfiguration[]
+}
+
+export const CreatorNotifications = () => {
+  const { userId } = useParams()
+  const [notificationSettings, setNotificationSettings] = useState<NotificationSettings[]>([])
+  const [project, setProject] = useState<any>(null)
+
+  useQuery(QUERY_CREATOR_NOTIFICATIONS_SETTINGS, {
+    variables: { userId },
+    onCompleted(data) {
+      setNotificationSettings(data?.settingsNotificationsCreatorGet.notificationSettings || [])
+      setProject(data?.settingsNotificationsCreatorGet.project || null)
+    },
+  })
+
+  const [updateNotificationSetting] = useMutation(MUTATION_UPDATE_CREATOR_NOTIFICATIONS_SETTINGS)
+
+  const getConfigValue = (type: string, name: string) => {
+    const setting = notificationSettings.find((s) => s.notificationType === type)
+    return setting?.configurations.find((c) => c.name === name)?.value
+  }
+
+  const getConfigId = (type: string, name: string) => {
+    const setting = notificationSettings.find((s) => s.notificationType === type)
+    return setting?.configurations.find((c) => c.name === name)?.id
+  }
```

Comment:
Are the values for "type" and "name"  preDefined ? 

If yes, let's create an enum with those values, and use it instead of passing a string value. It'll prevent any errors by typo in the future. We try to minimize the use to constant values as much as possible.